eventauth.go

```diff
@@ -1170,8 +1172,15 @@ func (m *membershipAllower) membershipAllowedSelf() error { // nolint: gocyclo
 		if m.oldMember.Membership == Invite {
 			return nil
 		}
+		return m.membershipFailed(
+			"sender cannot leave from this state",
+		)
+
+	default:
+		return m.membershipFailed(
+			"sender cannot set their own membership to %q", m.newMember.Membership,
```

Comment:
State that the membership type is unknown?

Reply:
In some cases, like `Invite` and `Ban`, it isn't really unknown.

Reply:
I suppose I can add special cases for those.